library/src/main/java/com/novoda/downloadmanager/RoomFile.java

```diff
@@ -4,23 +4,23 @@
 import android.arch.persistence.room.Entity;
 import android.arch.persistence.room.ForeignKey;
 import android.arch.persistence.room.Index;
-import android.arch.persistence.room.PrimaryKey;
 import android.support.annotation.NonNull;
 
 import static android.arch.persistence.room.ForeignKey.CASCADE;
 
 @SuppressWarnings("checkstyle:visibilitymodifier") // Accessors will add a lot of boilerplate code.
 @Entity(
+        primaryKeys = {"file_id", "batch_id"},
```

Comment:
Combination primary key so that a file identifier is unique per batch rather than globally.